src/Core/GraphQl/Type/FieldsBuilder.php

```diff
@@ -131,51 +134,56 @@ public function getCollectionQueryFields(string $resourceClass, ResourceMetadata
+        $deprecationReason = $operation->getDeprecationReason();
 
-        if ($fieldConfiguration = $this->getResourceFieldConfiguration(null, $description, $deprecationReason, $resourceType, $resourceClass, false, null, $mutationName, null)) {
-            $fieldConfiguration['args'] += ['input' => $this->getResourceFieldConfiguration(null, null, $deprecationReason, $resourceType, $resourceClass, true, null, $mutationName, null)];
+        if ($fieldConfiguration = $this->getResourceFieldConfiguration(null, $description, $deprecationReason, $resourceType, $resourceClass, false, $mutationName)) {
+            $fieldConfiguration['args'] += ['input' => $this->getResourceFieldConfiguration(null, null, $deprecationReason, $resourceType, $resourceClass, true, $mutationName)];
         }
 
-        $mutationFields[$mutationName.$resourceMetadata->getShortName()] = $fieldConfiguration ?? [];
+        $mutationFields[$mutationName.$shortName] = $fieldConfiguration ?? [];
 
         return $mutationFields;
     }
 
     /**
      * {@inheritdoc}
      */
-    public function getSubscriptionFields(string $resourceClass, ResourceMetadata $resourceMetadata, string $subscriptionName): array
+    public function getSubscriptionFields(string $resourceClass, Operation $operation, string $subscriptionName): array
     {
         $subscriptionFields = [];
-        $shortName = $resourceMetadata->getShortName();
+        $shortName = $operation->getShortName();
         $resourceType = new Type(Type::BUILTIN_TYPE_OBJECT, true, $resourceClass);
-        $description = $resourceMetadata->getGraphqlAttribute($subscriptionName, 'description', "Subscribes to the $subscriptionName event of a $shortName.", false);
-        $deprecationReason = $resourceMetadata->getGraphqlAttribute($subscriptionName, 'deprecation_reason', null, true);
+        $description = $operation->getDescription();
+        $deprecationReason = $operation->getDeprecationReason();
 
-        if ($fieldConfiguration = $this->getResourceFieldConfiguration(null, $description, $deprecationReason, $resourceType, $resourceClass, false, null, null, $subscriptionName)) {
-            $fieldConfiguration['args'] += ['input' => $this->getResourceFieldConfiguration(null, null, $deprecationReason, $resourceType, $resourceClass, true, null, null, $subscriptionName)];
+        if ($fieldConfiguration = $this->getResourceFieldConfiguration(null, $description, $deprecationReason, $resourceType, $resourceClass, false, $subscriptionName)) {
+            $fieldConfiguration['args'] += ['input' => $this->getResourceFieldConfiguration(null, null, $deprecationReason, $resourceType, $resourceClass, true, $subscriptionName)];
         }
 
         if (!$fieldConfiguration) {
             return [];
         }
 
-        $subscriptionFields[$subscriptionName.$resourceMetadata->getShortName().'Subscribe'] = $fieldConfiguration;
+        // TODO: 3.0 change this
+        if ('update_subscription' === $subscriptionName) {
```

Comment:
Why is it needed to do this?

Reply:
I don't want to break any tests, therefore names must stay the same. The `update` operation already exists and is a Mutation we need unique operation names.